src/test/java/guitests/PersonNewGuiTest.java

```diff
@@ -0,0 +1,121 @@
+        assertTrue(pandaWongCardHandle.isShowingGracePeriod("Adding"));
+        assertMatching(pandaWongCardHandle, pandaWong);
+
+        //Delete Person A
+        personListPanel.use_PERSON_DELETE_ACCELERATOR();
+        sleepForGracePeriod();
+        assertTrue(personListPanel.contains(pandaWong));
+
+        //Create Person B
+        addPersonDialog = personListPanel.clickNew();
+        Person ipMan = new PersonBuilder("IP", "Man")
+                .withStreet("Kungfu street 51").withCity("Beijing").withPostalCode("K4UWIN")
+                .withBirthday("01.01.1970").withGithubUsername("ipman").withTags(td.colleagues).build();
+        addPersonDialog.enterNewValues(ipMan).clickOk();
+
+        PersonCardHandle personCardHandle = personListPanel.navigateToPerson(ipMan);
+        assertTrue(personCardHandle.isShowingGracePeriod("Adding"));
+        sleepForGracePeriod();
+        assertNull(personListPanel.contains(pandaWong)); //Make sure Person A doesn't appear again.
+        assertTrue(personListPanel.contains(ipMan));
+    }
+
+    @Test
+    public void addPerson_multipleAdd() {
+
+        //Create Person A
+        EditPersonDialogHandle addPersonDialog = personListPanel.clickNew();
+
+        Person pandaWong = new PersonBuilder("Panda", "Wong")
+                .withStreet("Chengdu Panda Street").withCity("Chengdu").withPostalCode("PANDA")
+                .withBirthday("01.01.1979").withGithubUsername("panda").withTags(td.colleagues, td.friends).build();
+        addPersonDialog.enterNewValues(pandaWong).clickOk();
+
+        PersonCardHandle pandaWongCard = personListPanel.navigateToPerson(pandaWong);
+        assertTrue(pandaWongCard.isShowingGracePeriod("Adding"));
+        assertMatching(pandaWongCard, pandaWong);
+
+        //Create Person B
+        addPersonDialog = personListPanel.clickNew();
+        Person ipMan = new PersonBuilder("IP", "Man")
+                .withStreet("Kungfu street 51").withCity("Beijing").withPostalCode("K4UWIN")
+                .withBirthday("01.01.1970").withGithubUsername("ipman").withTags(td.colleagues).build();
+        addPersonDialog.enterNewValues(ipMan).clickOk();
+
+        PersonCardHandle ipManCard = personListPanel.navigateToPerson(ipMan);
+        assertTrue(ipManCard.isShowingGracePeriod("Adding"));
+        assertMatching(ipManCard, ipMan);
+
+        sleepForGracePeriod();
+
+        assertMatching(pandaWongCard, pandaWong);
+        assertMatching(ipManCard, ipMan);
+    }
+
+    @Test
+    public void addPerson_cancelDialog() {
+        EditPersonDialogHandle addPersonDialog = personListPanel.clickNew();
+
+        Person pandaWong = new PersonBuilder("Panda", "Wong")
+                .withStreet("Chengdu Panda Street").withCity("Chengdu").withPostalCode("PANDA")
+                .withBirthday("01.01.1979").withGithubUsername("panda").withTags(td.colleagues, td.friends).build();
+        addPersonDialog.enterNewValues(pandaWong).clickCancel();
```

Comment:
You may want to add a few more persons to our typical data class (but don't add them to the list). That way you can avoid copy pasting code like this many times. Remember to follow the same convention when naming persons.